pkg/oci/oci.go

```diff
@@ -281,22 +399,126 @@ func NewAuthClient(repository string, authOptions *AuthOptions) (*oras_auth.Clie
 // NewRepository creates a client to the remote repository identified by
 // image using the given auth options.
 func NewRepository(image string, authOpts *AuthOptions) (*remote.Repository, error) {
-	repository, err := GetRepositoryFromImage(image)
+	repository, err := getRepositoryFromImage(image)
 	if err != nil {
-		return nil, fmt.Errorf("get repository from image %q: %w", image, err)
+		return nil, fmt.Errorf("getting repository from image %q: %w", image, err)
 	}
 	repo, err := remote.NewRepository(repository)
 	if err != nil {
-		return nil, fmt.Errorf("create remote repository: %w", err)
+		return nil, fmt.Errorf("creating remote repository: %w", err)
 	}
 	repo.PlainHTTP = authOpts.Insecure
 	if !authOpts.Insecure {
-		client, err := NewAuthClient(repository, authOpts)
+		client, err := newAuthClient(repository, authOpts)
 		if err != nil {
-			return nil, fmt.Errorf("create auth client: %w", err)
+			return nil, fmt.Errorf("creating auth client: %w", err)
 		}
 		repo.Client = client
 	}
 
 	return repo, nil
 }
+
+func getImageListDescriptor(imageStore oras.ReadOnlyTarget, image string) (ocispec.Index, error) {
```

Comment:
```suggestion
func getImageListDescriptor(imageStore oras.ReadOnlyTarget, normalizedImage string) (ocispec.Index, error) {
```

From the codepath this function only gets normalized image names, so this is not an issue. The issue is when this function in the future gets called with a non normalized one.
So I think the least we should do to "catch" and future issues is to name the parameter that way (Or another way - you know the problems with naming things :D)

Now that I've thought about it a little more. Maybe we should introduce a new type. So the compiler always checks our usages of it. I just don't want headaches in the future xD. But I think this would be another PR

Reply:
I see that parameter is passed to `imageStore.Resolve` and it's called there `reference`, so I think it's fine to call it reference. I'm not sure introducing a new type is a good idea, perhaps it adds a lot of boilerplate to cast from one to the other.

Reply:
Yep, the boilerplate code is the disadvantage. And I also don't know how "annoying" or how much boilerplate would be needed.